### ISSUE TITLE
Test suite build failure

### ISSUE THREAD

**Issue by bergmark** [centromere/pipes-cacophony#2]
Noticed in the stackage build.

```
Preprocessing test suite 'test-pipes-cacophony' for pipes-cacophony-0.1.1...
[1 of 5] Compiling Instances        ( tests/Instances.hs, dist/build/test-pipes-cacophony/test-pipes-cacophony-tmp/Instances.o )
[2 of 5] Compiling HandshakeStates  ( tests/HandshakeStates.hs, dist/build/test-pipes-cacophony/test-pipes-cacophony-tmp/HandshakeStates.o )

tests/HandshakeStates.hs:25:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c4 d4 h4’
    In the second argument of ‘handshakeState’, namely ‘noiseNNI’
    In the expression:
      handshakeState "NN" noiseNNI "" Nothing Nothing Nothing Nothing

tests/HandshakeStates.hs:36:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c19 d19 h19’
    In the second argument of ‘handshakeState’, namely ‘noiseKNI’
    In the expression:
      handshakeState
        "KN" noiseKNI "" (Just initStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:47:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c5 d5 h5’
    In the second argument of ‘handshakeState’, namely ‘noiseNKI’
    In the expression:
      handshakeState
        "NK" noiseNKI "" Nothing Nothing (Just (snd respStatic)) Nothing

tests/HandshakeStates.hs:58:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c20 d20 h20’
    In the second argument of ‘handshakeState’, namely ‘noiseKKI’
    In the expression:
      handshakeState
        "KK"
        noiseKKI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        Nothing

tests/HandshakeStates.hs:69:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c6 d6 h6’
    In the second argument of ‘handshakeState’, namely ‘noiseNEI’
    In the expression:
      handshakeState
        "NE"
        noiseNEI
        ""
        Nothing
        Nothing
        (Just (snd respStatic))
        (Just (snd respEphemeral))

tests/HandshakeStates.hs:80:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c21 d21 h21’
    In the second argument of ‘handshakeState’, namely ‘noiseKEI’
    In the expression:
      handshakeState
        "KE"
        noiseKEI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        (Just (snd respEphemeral))

tests/HandshakeStates.hs:91:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c3 d3 h3’
    In the second argument of ‘handshakeState’, namely ‘noiseNXI’
    In the expression:
      handshakeState "NX" noiseNXI "" Nothing Nothing Nothing Nothing

tests/HandshakeStates.hs:102:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c22 d22 h22’
    In the second argument of ‘handshakeState’, namely ‘noiseKXI’
    In the expression:
      handshakeState
        "KX"
        noiseKXI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        Nothing

tests/HandshakeStates.hs:113:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c23 d23 h23’
    In the second argument of ‘handshakeState’, namely ‘noiseXNI’
    In the expression:
      handshakeState
        "XN" noiseXNI "" (Just initStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:124:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c24 d24 h24’
    In the second argument of ‘handshakeState’, namely ‘noiseINI’
    In the expression:
      handshakeState
        "IN" noiseINI "" (Just initStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:135:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c25 d25 h25’
    In the second argument of ‘handshakeState’, namely ‘noiseXKI’
    In the expression:
      handshakeState
        "XK"
        noiseXKI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        Nothing

tests/HandshakeStates.hs:146:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c26 d26 h26’
    In the second argument of ‘handshakeState’, namely ‘noiseIKI’
    In the expression:
      handshakeState
        "IK"
        noiseIKI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        Nothing

tests/HandshakeStates.hs:157:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c27 d27 h27’
    In the second argument of ‘handshakeState’, namely ‘noiseXEI’
    In the expression:
      handshakeState
        "XE"
        noiseXEI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        (Just (snd respEphemeral))

tests/HandshakeStates.hs:168:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c28 d28 h28’
    In the second argument of ‘handshakeState’, namely ‘noiseIEI’
    In the expression:
      handshakeState
        "IE"
        noiseIEI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        (Just (snd respEphemeral))

tests/HandshakeStates.hs:179:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c29 d29 h29’
    In the second argument of ‘handshakeState’, namely ‘noiseXXI’
    In the expression:
      handshakeState
        "XX" noiseXXI "" (Just initStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:190:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c30 d30 h30’
    In the second argument of ‘handshakeState’, namely ‘noiseIXI’
    In the expression:
      handshakeState
        "IX" noiseIXI "" (Just initStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:201:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c7 d7 h7’
    In the second argument of ‘handshakeState’, namely ‘noiseNI’
    In the expression:
      handshakeState
        "N" noiseNI "" Nothing Nothing (Just (snd respStatic)) Nothing

tests/HandshakeStates.hs:212:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c31 d31 h31’
    In the second argument of ‘handshakeState’, namely ‘noiseKI’
    In the expression:
      handshakeState
        "K"
        noiseKI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        Nothing

tests/HandshakeStates.hs:223:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c32 d32 h32’
    In the second argument of ‘handshakeState’, namely ‘noiseXI’
    In the expression:
      handshakeState
        "X"
        noiseXI
        ""
        (Just initStatic)
        Nothing
        (Just (snd respStatic))
        Nothing

tests/HandshakeStates.hs:234:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c2 d2 h2’
    In the second argument of ‘handshakeState’, namely ‘noiseNNR’
    In the expression:
      handshakeState "NN" noiseNNR "" Nothing Nothing Nothing Nothing

tests/HandshakeStates.hs:245:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c33 d33 h33’
    In the second argument of ‘handshakeState’, namely ‘noiseKNR’
    In the expression:
      handshakeState
        "KN" noiseKNR "" Nothing Nothing (Just (snd initStatic)) Nothing

tests/HandshakeStates.hs:256:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c8 d8 h8’
    In the second argument of ‘handshakeState’, namely ‘noiseNKR’
    In the expression:
      handshakeState
        "NK" noiseNKR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:267:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c34 d34 h34’
    In the second argument of ‘handshakeState’, namely ‘noiseKKR’
    In the expression:
      handshakeState
        "KK"
        noiseKKR
        ""
        (Just respStatic)
        Nothing
        (Just (snd initStatic))
        Nothing

tests/HandshakeStates.hs:278:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c9 d9 h9’
    In the second argument of ‘handshakeState’, namely ‘noiseNER’
    In the expression:
      handshakeState
        "NE"
        noiseNER
        ""
        (Just respStatic)
        (Just respEphemeral)
        Nothing
        Nothing

tests/HandshakeStates.hs:289:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c35 d35 h35’
    In the second argument of ‘handshakeState’, namely ‘noiseKER’
    In the expression:
      handshakeState
        "KE"
        noiseKER
        ""
        (Just respStatic)
        (Just respEphemeral)
        (Just (snd initStatic))
        Nothing

tests/HandshakeStates.hs:300:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c10 d10 h10’
    In the second argument of ‘handshakeState’, namely ‘noiseNXR’
    In the expression:
      handshakeState
        "NX" noiseNXR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:311:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c36 d36 h36’
    In the second argument of ‘handshakeState’, namely ‘noiseKXR’
    In the expression:
      handshakeState
        "KX"
        noiseKXR
        ""
        (Just respStatic)
        Nothing
        (Just (snd initStatic))
        Nothing

tests/HandshakeStates.hs:322:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c1 d1 h1’
    In the second argument of ‘handshakeState’, namely ‘noiseXNR’
    In the expression:
      handshakeState "XN" noiseXNR "" Nothing Nothing Nothing Nothing

tests/HandshakeStates.hs:333:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c0 d0 h0’
    In the second argument of ‘handshakeState’, namely ‘noiseINR’
    In the expression:
      handshakeState "IN" noiseINR "" Nothing Nothing Nothing Nothing

tests/HandshakeStates.hs:344:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c11 d11 h11’
    In the second argument of ‘handshakeState’, namely ‘noiseXKR’
    In the expression:
      handshakeState
        "XK" noiseXKR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:355:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c12 d12 h12’
    In the second argument of ‘handshakeState’, namely ‘noiseIKR’
    In the expression:
      handshakeState
        "IK" noiseIKR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:366:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c13 d13 h13’
    In the second argument of ‘handshakeState’, namely ‘noiseXER’
    In the expression:
      handshakeState
        "XE"
        noiseXER
        ""
        (Just respStatic)
        (Just respEphemeral)
        Nothing
        Nothing

tests/HandshakeStates.hs:377:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c14 d14 h14’
    In the second argument of ‘handshakeState’, namely ‘noiseIER’
    In the expression:
      handshakeState
        "IE"
        noiseIER
        ""
        (Just respStatic)
        (Just respEphemeral)
        Nothing
        Nothing

tests/HandshakeStates.hs:388:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c15 d15 h15’
    In the second argument of ‘handshakeState’, namely ‘noiseXXR’
    In the expression:
      handshakeState
        "XX" noiseXXR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:399:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c16 d16 h16’
    In the second argument of ‘handshakeState’, namely ‘noiseIXR’
    In the expression:
      handshakeState
        "IX" noiseIXR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:410:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c17 d17 h17’
    In the second argument of ‘handshakeState’, namely ‘noiseNR’
    In the expression:
      handshakeState
        "N" noiseNR "" (Just respStatic) Nothing Nothing Nothing

tests/HandshakeStates.hs:421:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c37 d37 h37’
    In the second argument of ‘handshakeState’, namely ‘noiseKR’
    In the expression:
      handshakeState
        "K"
        noiseKR
        ""
        (Just respStatic)
        Nothing
        (Just (snd initStatic))
        Nothing

tests/HandshakeStates.hs:432:3:
    Couldn't match expected type ‘Crypto.Noise.Cipher.Plaintext’
                with actual type ‘HandshakePattern c18 d18 h18’
    In the second argument of ‘handshakeState’, namely ‘noiseXR’
    In the expression:
      handshakeState
        "X" noiseXR "" (Just respStatic) Nothing Nothing Nothing
[3 of 5] Compiling Imports          ( tests/Imports.hs, dist/build/test-pipes-cacophony/test-pipes-cacophony-tmp/Imports.o )
```

**Comment by centromere** [comment]
Has this been fixed in 0.1.2?

**Comment by bergmark** [comment]
Seems like it, thanks!
